external/fv3fit/fv3fit/pytorch/cyclegan/discriminator.py

```diff
@@ -77,18 +77,19 @@ def __init__(
             raise ValueError("n_convolutions must be at least 1")
         # max_filters = min_filters * 2 ** (n_convolutions - 1), therefore
         min_filters = int(max_filters / 2 ** (n_convolutions - 1))
-        convs = [
-            ConvBlock(
+        # first convolutional block must not have instance normalization, so that the
+        # discriminator can use information about the mean and standard deviation of
+        # the input data (generated images)
+        convs = []
+        convs.append(
+            convolution(
                 in_channels=in_channels,
                 out_channels=min_filters,
-                convolution_factory=curry(convolution)(
-                    kernel_size=strided_kernel_size, stride=2,
-                ),
-                activation_factory=leakyrelu_activation(
-                    negative_slope=0.2, inplace=True
-                ),
+                kernel_size=strided_kernel_size,
+                stride=2,
             )
-        ]
+        )
+        convs.append(leakyrelu_activation(negative_slope=0.2, inplace=True)())
```

Comment:
So it looks like you have separated convs from activation functions? In this case, does the list `convs` still apply? Maybe it is now layers since you have a conv layer followed by a ReLU activation "layer".

Also, curious what the motivation for `in_place` is.

Reply:
Yes, I've separated it just because in a later PR I add a geographically-resolved bias module between the convolution operation and the activation function.

These could be renamed "modules", but I find it a bit more descriptive to call it convs, because it's still applying a series of convolutional layers. The conv, geographic bias, and activation modules together form a single convolutional layer with a geographically-dependent bias (at least, that's how I think about it).

`inplace` just makes it run a little faster if you know you aren't re-using the input somewhere else.

Reply:
It might make it more clear if I combine these 2-3 layers in a `nn.Sequential` container so it is indeed a list of convolutional layers - I'll try to remember to do this in a later PR / on my working branch.